### PR TITLE
Various bola fixes

### LINKED ISSUE #83577: bolas dont work when thrown
Reporting client version: 515.1639

## Issue Summary
Bolas break when the victim grabs a item

## Round ID:
[229678](https://scrubby.melonmesa.com/round/229678)

## Testmerges:
- [[TM ONLY] Hijacked Revision Date for v3 TGS Api](https://www.github.com/tgstation/tgstation/pull/83227)

## Reproduction:
Its easily dumb, throw a bola at someone, the bola victim picks a item out of their bag or inventory or pocket literally any storage, now the bola victim can move freely with no slowdown at all.

### PR DESCRIPTION
**Pull request by Das15**
## About The Pull Request
Fixes and rewrites bola ensnaring so it works properly. 
Fixed cult bolas leaving in-hand sprite when they ensnare non-cult user.
Fixes #83577 
## Why It's Good For The Game
Bolas were one of the main ways of stopping hulked criminals when security officer was not equipped with lethal weaponry. This PR makes bola slowdown apply as it should be. Also while looking at bolas I've noticed cult bolas leave in-hand sprite when ensnaring non-cult user, so I've fixed that as well. Big thanks to @GoblinBackwards for figuring out what was wrong with bolas.
## Changelog
:cl: MrDas
fix: Bolas now slowdown properly.
fix: Cult bolas no longer leave in-hand sprite when they ensnare to non-cult user.
/:cl:

### CODE REVIEW

File: code/game/objects/items/handcuffs.dm [hunk 1]
```diff
@@ -537,6 +537,10 @@
 	gender = NEUTER
 	///Amount of time to knock the target down for once it's hit in deciseconds.
 	var/knockdown = 0
+	///Used for ensnaring the target when hit in after_throw override
+	var/is_mob_hit = FALSE
```
Comment (SmArtKar) [inline code/game/objects/items/handcuffs.dm:541]: Is this variable really necessary? You always use it in conjunction with ensnare_mob_ref anyways

File: code/game/objects/items/handcuffs.dm [hunk 2]
```diff
@@ -546,21 +550,34 @@
...
+	is_mob_hit = TRUE
+	ensnare_mob_ref = WEAKREF(hit_atom)
+
+/obj/item/restraints/legcuffs/bola/after_throw(datum/callback/callback)
+	. = ..()
+	if (!is_mob_hit && !ensnare_mob_ref)
+		return
+	var/atom/ensnare_mob = ensnare_mob_ref.resolve()
+	if (is_mob_hit && !isnull(ensnare_mob))
+		ensnare(ensnare_mob)
```
Comment (SmArtKar) [inline code/game/objects/items/handcuffs.dm:563]: Same here, you can just check if ensnare_mob_ref is not null in the first check and ensure that the atom exists in the second one

File: code/game/objects/items/handcuffs.dm (hunk 1 above)
Comment (Das15) [reply code/game/objects/items/handcuffs.dm:541]: Yeah, now that I think about it... It is possible to just use weakref. Thanks for mentioning it.

File: code/game/objects/items/handcuffs.dm (hunk 2 above)
Comment (Das15) [reply code/game/objects/items/handcuffs.dm]: Yup, did just that

**Pull request merged**